nautobot_floor_plan/models.py

```diff
@@ -79,6 +81,37 @@ def get_svg(self, *, user, base_url):
         """Get SVG representation of this FloorPlan."""
         return FloorPlanSVG(floor_plan=self, user=user, base_url=base_url).render()
 
+    def save(self, *args, **kwargs):
+        """Override save in order to update any existing tiles."""
+        if not self.created:
+            super().save(*args, **kwargs)
+            return
+        # Get origin_seed pre/post values
+        initial_instance = self.__class__.objects.get(pk=self.pk)
+        x_initial = initial_instance.x_origin_seed
+        y_initial = initial_instance.y_origin_seed
+        x_updated = self.x_origin_seed
+        y_updated = self.y_origin_seed
+
+        super().save(**kwargs)
+
+        if x_initial != x_updated or y_initial != y_updated:
+            self.update_tile_origins(x_initial, x_updated, y_initial, y_updated)
+
+    def update_tile_origins(self, x_initial, x_updated, y_initial, y_updated):
+        """Update any existing tiles if axis_origin_seed was modified."""
+        tiles = self.tiles.all()
+        x_delta = x_updated - x_initial
+        y_delta = y_updated - y_initial
+
+        # should bulk_update these instead?
+        with transaction.atomic():
+            for tile in tiles:
+                tile.x_origin += x_delta
+                tile.y_origin += y_delta
+
+                tile.validated_save()
```

Comment:
Won't this cause problems if there's say an existing tile at both (1, 1) and (3, 1) and we change the origin from (1, 1) to (3, 1)? That is, we'd update the first tile to (3, 1) and the validated_save would complain because of the collision with the existing (not-yet-updated) tile already at (3, 1)?

I think it might be possible to avoid this issue by calling `order_by()` on `tiles` dependent on the signs of `x_delta` and `y_delta`?

Reply:
I thought I had already tested this but apparently my test wasn't accurate because your example was easily prove-able. Updating.